File: Sources/Runtime/Renderer/StateProviderView.swift
import SwiftUI
import Core

struct StateProviderView: View {
    let definition: ComponentDefinition
    @State private var localState = LocalState()

    var body: some View {
        Group {
            if let children = definition.children {
                ComponentRenderer(components: children)
            }
        }
        .environment(\.localState, localState)
        .onAppear {
            localState.initialize(from: definition.localState)
        }
    }
}

File: Android/melody-runtime/src/main/java/com/melody/runtime/renderer/StateProviderView.kt
package com.melody.runtime.renderer

import androidx.compose.runtime.*
import com.melody.core.schema.ComponentDefinition
import com.melody.runtime.state.LocalState

/**
 * Creates a local state scope for child components.
 * Port of iOS StateProviderView.swift.
 */
@Composable
fun StateProviderView(definition: ComponentDefinition) {
    val localState = remember { LocalState() }

    LaunchedEffect(definition) {
        localState.initialize(definition.localState)
    }

    CompositionLocalProvider(
        LocalLocalState provides localState
    ) {
        definition.children?.let { ComponentRenderer(components = it) }
    }
}
